src/size.rs

```diff
@@ -46,9 +55,16 @@ impl TextSize {
     ///
     /// This is equivalent to `TextSize::default()` or [`TextSize::MIN`],
     /// but is more explicit on intent.
+    #[inline]
     pub const fn zero() -> TextSize {
         TextSize(0)
     }
+
+    /// A size of one.
+    #[inline]
+    pub const fn one() -> TextSize {
```

Comment:
Strong :-1: on this, "size of one" doen't really makes sense at this level of abstraction, `TextSize::of('a')` does. For the code in rust-analyzer that works with utf16 ranges, the correct approach is to do this check at the level of `u32`/`usize`.

We **do** guarantee that the metric we use is "bytes of the utf-8 encoding", but the way to get to that raw representation is via casting to/from number.

Reply:
Would you accept `TextSize::ASCII`? There are two distinct cases in ra currently [[1]](https://github.com/rust-analyzer/rust-analyzer/blob/fedab39011e1225b580f5afce8607c8eedad77e1/crates/ra_ide_db/src/line_index.rs) [[2]](https://github.com/rust-analyzer/rust-analyzer/blob/f7b0135c64da13d64fb0add0c96e84289beb07b4/crates/ra_ide_db/src/line_index_utils.rs) that use `.to_usize()` to compare against `1`, being the size of ASCII text. I'd argue these would be worse of on clarity of intent if they were spelled `char_len > TextSize::of('a')`, because there's nothing special about `'a'` here, it's about one-byte versus many-byte UTF representation.

Or should these two cases be _more_ aggressive about avoiding `TextSize` and using `usize` instead?

While I can see the argument against `::one()` as being an arbitrary measure, I think `::ASCII` (the size of an ASCII character) is both objective and fits the need in the two above examples.

Reply:
There are *only* two cases in rust-analyzer that need this, so this is a strong positive evidence that any API here does not pull its weight. 

Additionally, I think that code should be rewritten to not use `TextSize` in the first place: it works for transcribing between utf8 and utf16, so using TextSize, which is specifically for utf8, is misusing the abstraction.

The code originally was written without white chars in mind at all, where text size made sense. Utf16 was added retroactively, and text size was just carried over.